db/memtable.cc

```diff
@@ -962,51 +963,64 @@ void MemTable::MultiGet(const ReadOptions& read_options, MultiGetRange* range,
 
   MultiGetRange temp_range(*range, range->begin(), range->end());
   if (bloom_filter_) {
-    std::array<Slice*, MultiGetContext::MAX_BATCH_SIZE> keys;
-    std::array<bool, MultiGetContext::MAX_BATCH_SIZE> may_match = {{true}};
-    autovector<Slice, MultiGetContext::MAX_BATCH_SIZE> prefixes;
+    bool whole_key =
+        !prefix_extractor_ || moptions_.memtable_whole_key_filtering;
+    std::array<Slice, MultiGetContext::MAX_BATCH_SIZE> bloom_keys;
+    std::array<bool, MultiGetContext::MAX_BATCH_SIZE> may_match;
+    std::array<size_t, MultiGetContext::MAX_BATCH_SIZE> range_indexes;
     int num_keys = 0;
     for (auto iter = temp_range.begin(); iter != temp_range.end(); ++iter) {
-      if (!prefix_extractor_) {
-        keys[num_keys++] = &iter->ukey_without_ts;
+      std::unique_ptr<FragmentedRangeTombstoneIterator> range_del_iter(
+          NewRangeTombstoneIterator(
+              read_options, GetInternalKeySeqno(iter->lkey->internal_key())));
+      if (range_del_iter != nullptr) {
+        iter->max_covering_tombstone_seq = std::max(
+            iter->max_covering_tombstone_seq,
+            range_del_iter->MaxCoveringTombstoneSeqnum(iter->lkey->user_key()));
+        // TODO: consider not counting these as Bloom hits to more closely
+        // match bloom_sst_hit_count
+        PERF_COUNTER_ADD(bloom_memtable_hit_count, 1);
+      } else if (whole_key) {
+        bloom_keys[num_keys] = iter->ukey_without_ts;
+        range_indexes[num_keys++] = iter.index();
       } else if (prefix_extractor_->InDomain(iter->ukey_without_ts)) {
```

Comment:
Was it wrong to do this without checking `memtable_whole_key_filtering`?

Reply:
Don't think so. The comment for memtable_prefix_bloom_size_ratio says the feature is disabled if both prefix_extractor and memtable_whole_key_filtering are not set. And if if both are set but key is not in domain, we would have just skipped the bloom filter and called GetFromTable.

Reply:
Like Anand said, just suboptimal, not wrong.